Jenkinsfile

```diff
@@ -340,6 +340,32 @@ def k8sTest(Map args = [:]) {
   }
 }
 
+/**
+* This method runs the GitHub action
+*/
+def githubAction(Map args = [:]) {
+  // We don't support branches/tags for the time being.
+  if (!isPR()) { return }
+  // Notify the status with a specific GitHub Check
+  withGithubNotify(context: args.context) {
+    // Provision a worker with the given labels. For the time being, let's provision a worker.
+    // We can revisit this to maybe use the async mode when triggering the github action.
+    withNode(labels: args.label, forceWorkspace: true) {
+      def result = githubWorkflowRun(
+        workflow: "macos-build",
+        ref: env.TARGET_BRANCH,
+          parameters: [
+            pr: env.CHANGE_ID,
+            path: args.directory,
+            command: args.command,
+            go_version: env.GO_VERSION,
+          ])
+      log(level: 'INFO', text: "Github build link: ${result.html_url}")
+      log(level: 'INFO', text: "Github build status: ${result.conclusion}")
```

Comment:
I saw the build worked, awesome!

![image](https://user-images.githubusercontent.com/2871786/144048816-d929f416-c399-4d12-9f17-98192e86c3a4.png)

I think we can now add a conclusion validation if not `success` then report an error? IIUC, `result.conclusion` contains `success` if it successfully ran.

So something like the below snippet could work:

```
if (!result.conclusion.equals('success') {
  error(''${args.context}: ${result.conclusion}. See ${result.html_url}")
}
```

What do you think?

OTOH, is the conclusion related to a `run` or a `step`? 

I see `steps.<step id>.conclusion` in https://docs.github.com/en/actions/learn-github-actions/contexts#steps-context versus `job.status` in https://docs.github.com/en/actions/learn-github-actions/contexts#job-context. The conclusion is the status for the whole run? Where can I see all the fields for the `result` object?